packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts

```diff
@@ -0,0 +1,294 @@
+                initialSummarizerDelayMs: 0,
+                summarizerClientElection: false,
+            },
+        },
+        gcOptions,
+    };
+
+    const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+        runtime.IFluidHandleContext.resolveHandle(request);
+
+    const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+        dataObjectFactory,
+        [
+            [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+        ],
+        undefined,
+        [innerRequestHandler],
+        runtimeOptions,
+    );
+
+    const settings = {
+        "Fluid.GarbageCollection.Test.Tombstone": "true",
+        "Fluid.GarbageCollection.TestOverride.SweepTimeoutMs": sweepTimeoutMs,
+    };
+    const configProvider = mockConfigProvider(settings);
+    const createContainer = async () => provider.createContainer(runtimeFactory, { configProvider });
+    const loadSummarizerAndContainer = async (summaryVersion?: string) => {
+        const absoluteUrl = await mainContainer.getAbsoluteUrl("");
+        return createSummarizerWithContainer(provider, absoluteUrl, runtimeFactory, { configProvider }, summaryVersion);
+    };
+    const summarize = async (summarizer: ISummarizer) => {
+        await provider.ensureSynchronized();
+        return summarizeNow(summarizer);
+    };
+
+    let mainContainer: IContainer;
+    let mainDataStore: TestDataObject;
+
+    beforeEach(async function() {
+        provider = getTestObjectProvider({ syncSummarizer: true });
+        if (provider.driver.type !== "local") {
+            this.skip();
+        }
+        mainContainer = await createContainer();
+        mainDataStore = await requestFluidObject<TestDataObject>(mainContainer, "default");
+        await waitForContainerConnection(mainContainer);
+    });
+
+    const createUnreferencedDataStore = async () => {
+        const handleKey = "handle";
+        const testDataObject = await dataObjectFactory.createInstance(mainDataStore.containerRuntime);
+        const testDataObjectId = testDataObject.id;
+
+        // Reference a datastore and summarize
+        mainDataStore._root.set(handleKey, testDataObject.handle);
+        const {
+            container: summarizingContainer1,
+            summarizer: summarizer1,
+        } = await loadSummarizerAndContainer();
+        await summarize(summarizer1);
+
+        // Unreference a datastore and summarize
+        mainDataStore._root.delete(handleKey);
+        const summaryVersion = (await summarize(summarizer1)).summaryVersion;
+        testDataObject._root.set("send while unreferenced", "op");
+        await provider.ensureSynchronized();
+        mainContainer.close();
+        summarizingContainer1.close();
+
+        return {
+            testDataObjectId,
+            summaryVersion,
+        };
+    };
+
+    // If this test starts failing due to runtime is closed errors try first adjusting `sessionExpiryTimeoutMs` above
```

Comment:
Can you add more details to this comment - Why would it fail and what adjustment needs to be done? Does this mean the test is flaky?

Reply:
Got rid of` sessionExpiry` so this test is less flakey, anything with sweep timeout/timers makes the test flakey